micrometer-core/src/main/java/io/micrometer/core/instrument/binder/kafka/KafkaConsumerMetrics.java

```diff
@@ -211,10 +215,12 @@ int kafkaMajorVersion(Tags tags) {
                             String version = (String) mBeanServer.getAttribute(new ObjectName(JMX_DOMAIN + ":type=app-info,client-id=" + clientId.getValue()), "version");
                             return Integer.parseInt(version.substring(0, version.indexOf('.')));
                         } catch (Throwable e) {
-                            return -1; // should never happen
+                            // this can happen during application bootstrapping
+                            // in this case, JMX bean is not available yet and version cannot be determined yet
+                            return null;
```

Comment:
I noticed the issue of this block being used and the result cached also, but returning `null` here will result in a NPE.